test/rekt/resources/account_role/role.go

```diff
@@ -46,6 +48,26 @@ func WithRole(role string) manifest.CfgFn {
 	}
 }
 
+func WithRules(rules ...rbacv1.PolicyRule) manifest.CfgFn {
+	return func(cfg map[string]interface{}) {
+		if _, set := cfg["rules"]; !set {
+			cfg["rules"] = []map[string]interface{}{}
+		}
+
+		for _, resource := range rules {
+			elem := map[string]interface{}{
+				"apiGroups": resource.APIGroups,
+				"resources": resource.Resources,
+				"verbs":     resource.Verbs,
+				// TODO: skip for now
+				//"resourceNames":   resource.ResourceNames,
+				//"nonResourceURLs": resource.NonResourceURLs,
```

Comment:
I didn't want to add these for now in order not to waste time and focus on getting feedback on the general idea of the PR.

Would we ever need these? Happy to add code that handles these too, if necessary.